.github/workflows/TEMPLATE_provider.yml

```diff
@@ -1,41 +1,64 @@
-# Replace all occurrences of "PROVIDER" with the provider name
-# Replace all occurrences of "provider" with the folder name of the provider (e.g. `nats`, `http-client`, etc)
-name: PROVIDER
+# Replace all occurrences of "PROVIDER" with the provider name (e.g. HTTPCLIENT)
+# Replace workflow triggers with provider with the provider name, should be the same as the folder path
+name: PROVIDER-RELEASE
 
 on:
   push:
-    branches: [ main ]
-    paths:
-    - "provider/**"
-  pull_request:
-    branches: [ main ]
-    paths:
-    - "provider/**"
+    tags:
+    - 'provider-v*'
 
 env:
   CARGO_TERM_COLOR: always
   working-directory: ./provider
+  WASH_ISSUER_KEY: ${{ secrets.WASMCLOUD_ACCOUNT_OFFICIAL }}
+  WASH_SUBJECT_KEY: ${{ secrets.WASMCLOUD_PROVIDER }}
+  WASH_REG_USER:  ${{ secrets.AZURECR_PUSH_USER }}
+  WASH_REG_PASSWORD:  ${{ secrets.AZURECR_PUSH_PASSWORD }}
+  REVISION: ${{ github.run_number }}
 
 jobs:
-  cargo_check:
+  rust_check:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
-    - name: Build
-      run: cargo build --verbose
-      working-directory: ${{env.working-directory}}
-    - name: Run tests
-      run: cargo test --verbose -- --nocapture
-      working-directory: ${{env.working-directory}}
-    - name: Check fmt
-      run: cargo fmt -- --check
-      working-directory: ${{env.working-directory}}
+    - id: rust-check-action
+      uses: wasmcloud/common-actions/rust-check@main
+      with:
+        working-directory: ${{ env.working-directory }}
 
-  clippy_check:
+  github_release:
+    if: startswith(github.ref, 'refs/tags/') # Only run on tag push
+    needs: rust_check
+    runs-on: ubuntu-latest
+    steps:
+    - name: Create Release
+      id: create_release
+      uses: actions/create-release@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        tag_name: ${{ github.ref }}
+        release_name: Release ${{ github.ref }}
+        draft: false
+        prerelease: true
+
+  artifact_release:
+    if: startswith(github.ref, 'refs/tags/') # Only run on tag push
+    needs: github_release
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - run: rustup component add clippy
-      - name: Run cargo clippy
-        run: cargo clippy --all-features
-        working-directory: ${{env.working-directory}}
+      - uses: wasmcloud/common-actions/install-wash@main
+      - uses: wasmcloud/common-actions/install-cross@main
+      - name: Build full provider archive
+        run: make par-full
+        working-directory: ${{ env.working-directory }}
+      # Push artifact to https://AZURECR/provider:VERSION
```

Comment:
can push artifact move to a common action?

Reply:
Yes, the way that would have to work is by giving it an input for the path to the artifact (e.g. `httpclient/build/httpclient.par.gz`) and a registry URL. Let me see what I can get working